Essentials/src/main/java/com/earth2me/essentials/utils/VersionUtil.java

```diff
@@ -46,17 +62,33 @@ public static BukkitVersion getServerBukkitVersion() {
     }
 
     public static SupportStatus getServerSupportStatus() {
-        try {
-            Class.forName("net.minecraftforge.common.MinecraftForge");
-            return SupportStatus.UNSTABLE;
-        } catch (final ClassNotFoundException ignored) {
-        }
+        if (supportStatus == null) {
+            for (String clazz : dangerousServerClasses) {
+                try {
+                    Class.forName(clazz);
+                    return supportStatus = SupportStatus.DANGEROUS;
+                } catch (final ClassNotFoundException ignored) {
+                }
+            }
 
-        if (!supportedVersions.contains(getServerBukkitVersion())) {
-            return SupportStatus.OUTDATED;
-        }
+            try {
+                Class.forName("net.minecraftforge.common.MinecraftForge");
+                return supportStatus = SupportStatus.UNSTABLE;
+            } catch (final ClassNotFoundException ignored) {
+            }
+
+            // Check for non-nms Bukkit Implis
+            if (ReflUtil.getNMSClass("MinecraftServer") == null) {
+                return supportStatus = SupportStatus.UNSTABLE;
```

Comment:
The `UNSTABLE` message is for Forge hybrid servers, but this will trigger on other impls like Glowstone.